src/canvasdrawer.js

```diff
@@ -0,0 +1,1151 @@
+            }
+        }
+
+        if (useSketch) {
+            if (sketchScale) {
+                if (this.viewport.getRotation(true) % 360 !== 0) {
+                    this._offsetForRotation({
+                        degrees: this.viewport.getRotation(true),
+                        useSketch: false
+                    });
+                }
+                if (tiledImage.getRotation(true) % 360 !== 0) {
+                    this._offsetForRotation({
+                        degrees: tiledImage.getRotation(true),
+                        point: this.viewport.pixelFromPointNoRotate(
+                            tiledImage._getRotationPoint(true), true),
+                        useSketch: false
+                    });
+                }
+            }
+            this.blendSketch({
+                opacity: tiledImage.opacity,
+                scale: sketchScale,
+                translate: sketchTranslate,
+                compositeOperation: tiledImage.compositeOperation,
+                bounds: bounds
+            });
+            if (sketchScale) {
+                if (tiledImage.getRotation(true) % 360 !== 0) {
+                    this._restoreRotationChanges(false);
+                }
+                if (this.viewport.getRotation(true) % 360 !== 0) {
+                    this._restoreRotationChanges(false);
+                }
+            }
+        }
+
+        if (!sketchScale) {
+            if (this.viewport.getRotation(true) % 360 === 0 &&
+                tiledImage.getRotation(true) % 360 === 0) {
+                if(this.viewer.viewport.getFlip()) {
+                    this._flip();
+                }
+            }
+        }
+
+        this._drawDebugInfo( tiledImage, lastDrawn );
+
+        // Fire tiled-image-drawn event.
+        if( this.viewer ){
+            /**
+                * Raised when a tiled image is drawn to the canvas. Only valid
+                * for webgl and canvas drawers.
+                *
+                * @event tiled-image-drawn
+                * @memberof OpenSeadragon.Viewer
+                * @type {object}
+                * @property {OpenSeadragon.Viewer} eventSource - A reference to the Viewer which raised the event.
+                * @property {OpenSeadragon.TiledImage} tiledImage - Which TiledImage is being drawn.
+                * @property {Array} tiles - An array of Tile objects that were drawn.
+                * @property {?Object} userData - Arbitrary subscriber-defined object.
+                */
+            this.viewer.raiseEvent( 'tiled-image-drawn', {
+                tiledImage: tiledImage,
+                tiles: lastDrawn,
+            });
+        }
+    }
+
+    /**
+         * @private
+         * @inner
+         * This function converts the given point from to the drawer coordinate by
+         * multiplying it with the pixel density.
+         * This function does not take rotation into account, thus assuming provided
```

Comment:
Should this function have `NoRotate` in its title to make that explicit?

Reply:
This wasn't changed from the current implementation. It's technically a public API so I think renaming it would be a breaking change. I guess we'd have to deprecate it first...

https://github.com/openseadragon/openseadragon/blob/6abcc76a93ee3369267908f0a80641dc171440ed/src/drawer.js#L173-L181